android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -493,6 +438,41 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+    private List<WalletListItemModel> createAccountList(AccountInfo[] accountInfos,
+            double assetPrice, HashMap<String, Double> nativeAssetsBalances,
+            HashMap<String, HashMap<String, Double>> blockchainTokensBalances) {
+        List<WalletListItemModel> walletListItemModelList = new ArrayList<>();
+        for (AccountInfo accountInfo : accountInfos) {
+            final String accountAddressLower = accountInfo.address.toLowerCase(Locale.ENGLISH);
+            double thisAccountBalance = getAccountBalance(
+                    nativeAssetsBalances, blockchainTokensBalances, accountAddressLower);
+            final String fiatBalanceString =
+                    String.format(Locale.ENGLISH, "$%,.2f", assetPrice * thisAccountBalance);
+            final String cryptoBalanceString =
+                    String.format(Locale.ENGLISH, "%.4f %s", thisAccountBalance, mAsset.symbol);
+
+            // If NFT, only show the account that owns
+            // it (i.e. balance = 1)
+            if (mAsset.isNft && thisAccountBalance != 1.) continue;
```

Comment:
Naive question: Is this something that may happen? Can an NFT land in the AssetDetailActivity?

Reply:
Don't think though haven't tested after NFT tab PR as it is still going, though probably be a good place to refactor with `Include multichain support` Issue? There could be similar places to clean up.

Reply:
Yes, absolutely.
Adding a note to double-check this in the multichain support issue.